### PR TITLE
I/O refactor

### PR DESCRIPTION
**Pull request by harshkpatel**
A refactor for the I/O so it abides clean architecture and SOLID. I think the changes are good, but I would like feedback in one area, which is how I am passing the use case interactor. I decided to take it as an input in both and input and output interactor, and while it makes sense for it to be created externally for output, I can see a reason for creating it locally for input. I ended up going with creating it externally for both interactors as it made more sense since both input and output need to be connected, but I think either case has the argument. I would like some feedback for what you guys think and feel free to point out if there are any glaring issues with how I decided to split everything up and if there are any dependencies or if any class is doing too much.

### CODE REVIEW

**Review by derrcodes — APPROVED** [review]
The refactor seems really good and organized and I think it adheres to clean architecture and SOLID principles. I think in regards to you questions, as long as you can justify a choice over the other, which in this case you did by mentioning that input and output are connected. I don't see any glaring issues and I personally think the new implementation for the I/O is really clean so great job on the refactor.

File: src/main/java/Interactors/OutputInteractor.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * This function deals with all the states that need to be updated periodically based on the current state of the game
+     * @param state: the current state that game is in
+     */
+    public void updateLogicStates(String state){
+        StringBuilder currString = new StringBuilder();
+        switch (state) {
+            case "MainTree":
```
Comment (Leuxll) [inline src/main/java/Interactors/OutputInteractor.java:47]: Should we use a switch statement here, isn't that a code smell? Also, the method is kinda long, which would be a code smell, more specifically a bloater, wouldn't it be better if we refactor these out to different methods and then after that just call those methods in this method?

**Review by Leuxll — APPROVED** [review]
Overall good job in refactoring the logic out of the entities into an Interactor more specifically an InputInteractor.

**Comment by harshkpatel** [comment]
I accidentally made commits to the wrong branch ignore all those commits

File: src/main/java/Interactors/OutputInteractor.java (hunk 1 above)
Comment (benxu3) [reply src/main/java/Interactors/OutputInteractor.java:47]: I agree with this feedback, it looks like the code follows the same logic for each case. Could you try to refactor the switch statement by creating a helper method following the logic in each case body and which takes as input the parameters which change?

File: src/main/java/Entities/Output.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * The constructor for this class, it pre-defines all the key-value pairs in the hashmap
+     * note: some values are null as those values do not present any output to the user and are simply processing stages
+     * note: some keys start off as empty string as those change based on logic, so empty strings are used as placeholders
+     * note: the Value with respect to each key represents the output giving the option for the user to choose what they want to do
+     * i.e. the output string value is to determine the next state from the current state
+     */
+    public Output(){
+        this.stateOutput = new HashMap<>();
+        this.stateOutput.put("InitialMenu", "Welcome to Monopoly++, Would you like to ");
```
Comment (benxu3) [inline src/main/java/Entities/Output.java:25]: Would it be better to store these strings in a txt file and have a helper which reads each line and adds it to the class? I think you may need to create an interface and whatnot but I'm not sure if this is the best way to populate the HashMap.

**Review by benxu3 — COMMENTED** [review]
Overall looks good, thanks for refactoring the code. Take a look at my comments, I think you may be able to implement certain things better.

File: src/main/java/Interactors/OutputInteractor.java (hunk 1 above)
Comment (harshkpatel) [reply src/main/java/Interactors/OutputInteractor.java:50]: I address the comment about splitting this up, but with regards to the code smell about switch statements, that will be dealt with in a later refactor with regards to actually creating an abstract class for the different states and the child classes which all output their respective state in a string

File: src/main/java/Entities/Output.java (hunk 2 above)
Comment (harshkpatel) [reply src/main/java/Entities/Output.java]: Made a new class and moved the string into a file

**Pull request merged**